Components/register

RayFreedom88 opened the pull request:
- добавлена страница регистрации
-  в index.css добавлены цветовые переменные
- дополнен компонент Login и Input
-  в файле стилей Login заменены цвета на переменные

RayFreedom88 committed:
feat: добавить функционал в Login, Input

- в компоненте Login добавил валидацию и добавил переключение иконки при скрытии/показе пароля
-  также добавлены три переменные в index.css

RayFreedom88 committed:
feat: дополнить стили index.css, input.module.css, button.module.css

RayFreedom88 committed:
fix: заменить в Login.module.css хадкор цвета на переменные

RayFreedom88 committed:
feat: добавить new-components/regiser.tsx

tarassov commented on src/new-components/register/register.tsx:
```diff
@@ -0,0 +1,165 @@
+		if (e.target.value) {
+			setIsNameError(!validateName(e.target.value));
+		} else {
+			setIsMailError(false);
```
а тут разве не setIsNameError должно быть?

RayFreedom88 replied:
да, все верно) не углядел)

RayFreedom88 replied:
этот момент исправил в следующем коммите

tarassov commented on src/new-components/register/register.tsx:
```diff
@@ -0,0 +1,165 @@
+		}
+	};
+
+	const handlePasswordBlur = (e: React.FocusEvent<HTMLInputElement>) => {
+		if (e.target.value.length <= 5) {
+			setIsPasswordError(true);
+		} else {
+			setIsPasswordError(false);
+		}
+	};
```
мне кажется, тут тогда лучше еще в ошибки писать чем пароль не угодил, а то сразу и не ясно чего он красным его подсветил

RayFreedom88 replied:
добавил вывод текста ошибки

RayFreedom88 committed:
feat: добавить в компонент Input вывод текста ошибки

RayFreedom88 committed:
update: подтянуть изменения для решения конфликта в ПР

jack-theripper commented on src/new-components/register/register.tsx:
```diff
@@ -0,0 +1,169 @@
+		setPasswordVisible((state) => !state);
+	};
+
+	const handleChange = (e: { target: HTMLInputElement }) => {
```
Все-таки типизацию эвента лучше сделай через дженерик FormEvent :-)))

RayFreedom88 replied:
Как скажете)

jack-theripper commented on src/new-components/register/register.tsx:
```diff
@@ -0,0 +1,169 @@
+const Register: FC = () => {
+	const dispatch = useDispatch();
+
+	const state = useSelector<any, AuthForm>((state) => ({
```
Типизация состояния для страницы логин отличается от регистрации

jack-theripper commented on src/new-components/register/register.tsx:
```diff
@@ -0,0 +1,169 @@
+		...state.auth,
+	}));
+
+	const [formValue, setFormValue] = useState({
+		name: "",
+		email: "",
+		password: "",
+	});
```
т.к. дополнительно вводишь еще локальный стейт поэтому не получаешь ошибок от ts

jack-theripper commented on src/new-components/login/Login.tsx:
```diff
@@ -59,6 +60,35 @@ const Login: React.FC = () => {
+		}
+	};
+
+	const handlePasswordBlur = (e: React.FocusEvent<HTMLInputElement>) => {
+		if (e.target.value.length <= 5) {
+			setIsPasswordError(true);
+			SetPasswordErrorText("Пароль должен быть не менее 6 символов");
+		} else {
+			setIsPasswordError(false);
+		}
+	};
```
При входе на сайт не нужно  длину пароля проверять ведь

RayFreedom88 replied:
тоже верно)

jack-theripper commented on src/new-components/input/input.tsx:
```diff
@@ -1,36 +1,62 @@
+import styles from "./input.module.css";
 
 interface FormInputProps {
+	name?: string;
```
Зачем ему name если значения в стейте хранятся)

RayFreedom88 replied:
чтобы по феншую было)

jack-theripper commented on src/new-components/input/input.tsx:
```diff
@@ -1,36 +1,62 @@
+	errorText,
+	children,
+}) => {
+	const errorToRender = useMemo(
```
Мемоизация нужна для промежуточных вычислений, для верстки - не нужно

RayFreedom88 replied:
если убрать, как ты говоришь, то ошибка выводиться не будет! Поэтому useMemo остается!

RayFreedom88 committed:
feat: добавить подробный вывод ошибок при валидации

- fix(Login): убрал валидацию пароля
- fix(Register): переделана типизация state
- feat(Login): добавлен подробный  вывод ошибки при валидации
- feat(Register) : добавлен подробный вывод ошибок при валидации
